app/controllers/api/functions.php

```diff
@@ -442,6 +451,57 @@
         ]), Response::MODEL_RUNTIME_LIST);
     });
 
+App::get('/v1/functions/specifications')
+    ->groups(['api', 'functions'])
+    ->desc('Get available function runtime specifications')
+    ->label('scope', 'functions.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'getSpecifications')
+    ->label('sdk.description', '/docs/references/functions/get-specifications.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_SPECIFICATION_LIST)
+    ->inject('response')
+    ->inject('plan')
+    ->action(function (Response $response, array $plan) {
+        $allRuntimeSpecs = Config::getParam('runtime-specifications', []);
+        $plans = Config::getParam('plans', []);
+
+        $runtimeSpecs = [];
+        foreach ($allRuntimeSpecs as $key => $spec) {
+            // Don't perform any billing calculations for CE
+            if (empty($plans)) {
+                $spec['enabled'] = true;
+            } else {
+                // Show plan names on specs
+                foreach ($plans as $billingPlan) {
+                    if (key_exists('runtimeSpecifications', $billingPlan) && in_array($key, $billingPlan['runtimeSpecifications'])) {
+                        if ($billingPlan['$id'] !== 'tier-0') {
+                            $spec['plan'] = $billingPlan['name'];
+                        }
+                        break;
+                    }
+                }
```

Comment:
Can this be abstracted to cloud repository, by using triggers? Like we used to use for ZDT.

Reply:
Do you know how and where we create hooks in Cloud/